packages/x-components/src/x-modules/history-queries/components/my-history.vue

```diff
@@ -1,58 +1,78 @@
 <template>
-  <BaseSuggestions
-    :suggestions="historyQueries"
-    class="x-my-history-queries"
-    data-test="my-history-queries"
-    :animation="animation"
-  >
-    <template #default="{ suggestion, index }">
-      <!--
+  <ul v-if="historyQueries.length > 0" class="x-my-history x-list">
```

Comment:
i'd change this hardcoded `ul` to a dynamic component so we can add an `animation` component to the container list.

```vue
    <component :is="animation" v-if="hasHistoryQueries" class="x-my-history x-list" tag="ul">
```

Update the `animation` `prop` accordingly:

```typescript
@Prop({ default: 'ul' })
protected animation!: Vue | string;
```

I'd refactor the `historyQueries.length > 0` check to a computed property:

```typescript
/**
     * The `hasHistoryQueries` computed property is a flag representing if there are history queries
     * stored.
     *
     * @returns True if there are history queries; false otherwise.
     *
     * @internal
     */
    protected get hasHistoryQueries(): boolean {
      return !isArrayEmpty(this.historyQueries);
    }
```